Dockerfile

```diff
@@ -1,19 +1,29 @@
 ARG PHP_VERSION=7.2
+ARG COMPOSER_VERSION=1.8
 
+FROM composer:${COMPOSER_VERSION}
 FROM php:${PHP_VERSION}-cli
 
+RUN set -eux; \
+    if [ $(php -r "echo PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION;") = "7.3" ]; \
+    then \
+        pecl install xdebug-beta; \
+    else \
+        pecl install xdebug; \       
+    fi && \
+    docker-php-ext-enable xdebug
+
 RUN apt-get update && \
     apt-get install -y git zip unzip && \
-    pecl install mongodb && docker-php-ext-enable mongodb && \
-    pecl install xdebug && docker-php-ext-enable xdebug
-
-RUN curl -sS https://getcomposer.org/installer | php \
-    && mv composer.phar /usr/local/bin/ \
-    && ln -s /usr/local/bin/composer.phar /usr/local/bin/composer
+    pecl install mongodb && docker-php-ext-enable mongodb
 
-ADD composer.json /code/composer.json
-ADD composer.lock /code/composer.lock
-RUN cd /code && composer install --prefer-source --no-interaction
+COPY --from=0 /usr/bin/composer /usr/local/bin/composer
 
 WORKDIR /code
-ENV PATH="~/.composer/vendor/bin:./vendor/bin:${PATH}"
+
+ADD composer.json composer.json
+
+# composer.lock isn't exists in git repository
+#ADD composer.lock composer.lock 
```

Comment:
You can remove these lines :)

Reply:
Ok

Reply:
Please, in free time review this PR https://github.com/jenssegers/laravel-mongodb/pull/1614 ;)